diff --git a/chrome/common/BUILD.gn b/chrome/common/BUILD.gn
index 7ca361112ae6553b5d55c0118868b6592466e466..0319cfee636d9f560166946e8b0c9cb8b7186418 100644
--- a/chrome/common/BUILD.gn
+++ b/chrome/common/BUILD.gn
@@ -187,7 +187,7 @@ static_library("common") {
     "tts_messages.h",
     "tts_utterance_request.cc",
     "tts_utterance_request.h",
-    "url_constants.cc",
+    "//brave/common/url_constants.cc",
     "url_constants.h",
     "web_application_info.cc",
     "web_application_info.h",
@@ -533,6 +533,13 @@ static_library("non_code_constants") {
     "env_vars.h",
   ]
 
+  if (brave_chromium_build && is_win) {
+    sources -= [ "chrome_icon_resources_win.h" ]
+    sources += [
+      "//brave/chromium_src/chrome/common/chrome_icon_resources_win.h",
+    ]
+  }
+
   deps = [
     ":buildflags",
     ":version_header",
@@ -564,6 +571,15 @@ static_library("constants") {
     "pref_names.h",
   ]
 
+  if (brave_chromium_build && is_official_build) {
+    sources += [
+      "//brave/common/brave_paths_mac.mm",
+      "//brave/common/brave_paths_mac.h",
+      "//brave/common/brave_channel_info_linux.cc",
+      "//brave/common/brave_channel_info_linux.h",
+    ]
+  }
+
   public_deps = [
     ":buildflags",
     ":non_code_constants",
