diff --git a/third_party/WebKit/Source/modules/bluetooth/NavigatorBluetooth.cpp b/third_party/WebKit/Source/modules/bluetooth/NavigatorBluetooth.cpp
index 7d425055a14c210942e0b6e5d3b6719d4dcc898e..59226dcc911943f63e07e938e77d902c14ad8019 100644
--- a/third_party/WebKit/Source/modules/bluetooth/NavigatorBluetooth.cpp
+++ b/third_party/WebKit/Source/modules/bluetooth/NavigatorBluetooth.cpp
@@ -20,7 +20,11 @@ NavigatorBluetooth& NavigatorBluetooth::From(Navigator& navigator) {
 }
 
 Bluetooth* NavigatorBluetooth::bluetooth(Navigator& navigator) {
+#if defined(MUON_CHROMIUM_BUILD)
   return NavigatorBluetooth::From(navigator).bluetooth();
+#else 
+  return nullptr;
+#endif
 }
 
 Bluetooth* NavigatorBluetooth::bluetooth() {
